presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/CreateEmptyPartitionProcedure.java

```diff
@@ -70,14 +73,16 @@
     private final HiveMetastoreClosure metastore;
     private final LocationService locationService;
     private final JsonCodec<PartitionUpdate> partitionUpdateJsonCodec;
+    private final ConnectorAccessControl accessControl;
 
     @Inject
-    public CreateEmptyPartitionProcedure(TransactionalMetadataFactory hiveMetadataFactory, HiveMetastore metastore, LocationService locationService, JsonCodec<PartitionUpdate> partitionUpdateCodec)
+    public CreateEmptyPartitionProcedure(TransactionalMetadataFactory hiveMetadataFactory, HiveMetastore metastore, LocationService locationService, JsonCodec<PartitionUpdate> partitionUpdateCodec, ConnectorAccessControl accessControl)
```

Comment:
This is not enough. Access to a table can be controlled on system level access control. 

Actually I think we need to pass something that represents top `AccessControl`. Then we need different security context too.

@electrum Any thoughts?

Reply:
Yes, @Praveen2112 is changing this so that procedure calls are injected with a `ConnectorAccessControl` facade that does the connector and system access checks.

Reply:
Nice trick.